File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        count = 0

        for num in nums:
            local_count = 1
            if num-1 not in nums:
                while num + 1 in nums:
                    local_count += 1
                    num += 1
                count = max(count, local_count)

        return count

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counts;
        for (int num : nums) {
            counts[num] += 1;
        }

        vector<pair<int, int>> arr;
        for (auto& pair : counts) {
            arr.push_back({pair.second, pair.first});
        }
        sort(arr.begin(), arr.end());

        vector<int> res;

        for (int i = arr.size() - 1; k > 0; --k, --i) {
            res.push_back(arr[i].second);
        }

        return res;

    }
};
